BeeSwift/Gallery/GalleryViewController.swift

```diff
@@ -408,25 +406,31 @@ class GalleryViewController: UIViewController, UICollectionViewDelegateFlowLayou
         self.updateDeadbeatHeight()
         self.lastUpdated = Date()
         self.updateLastUpdatedLabel()
-        if self.goals.count == 0 {
-            self.noGoalsLabel.isHidden = false
-            self.collectionView?.isHidden = true
-        } else {
-            self.noGoalsLabel.isHidden = true
-            self.collectionView?.isHidden = false
-        }
-        let searchItem = UIBarButtonItem(barButtonSystemItem: .search, target: self, action: #selector(self.searchButtonPressed))
-        self.navigationItem.leftBarButtonItem = searchItem
+        self.navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .search, target: self, action: #selector(self.searchButtonPressed))
     }
 
     func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
         return ServiceLocator.versionManager.lastChckedUpdateState() == .UpdateRequired ? 0 : self.filteredGoals.count
     }
     
     func numberOfSections(in collectionView: UICollectionView) -> Int {
-        return 1
+        [filteredGoals.isEmpty, goals.isEmpty].contains(true) ? 0 : 1
     }
     
+    private func setBackgroundView(filteredGoals: [Goal], goals: [Goal]) {
+        var view: UIView? {
+            switch (filteredGoals.isEmpty, goals.isEmpty) {
+            case (true, false):
+                return self.makeViewForEmptyCollection(when: .noGoalsMatchingFilter)
```

Comment:
[Do] Remove the enum and the multiple views, just pass in a string here.

I've marked this as a Do, but I'm open to being convinced otherwise. At the moment I don't see the benefit to justify the additional complexity.

Reply:
Currently `collectionView?.backgroundView` is being set to a UIView which is just a container, containing a BSLabel with the reason no goals are shown, and, below it, another UIView acting as a spacer.

The intent was to show the text/label "above the fold" for better visibility and to not have it sit in the middle, which would be covered by the MBProgressHUD activity indicator.

Your proposal is just to set the backgroundView to the label directly?

Reply:
I'm happy either to set it to the view, or the label directly. I just don't think we get much from having the `NoGoalReason` enum - `makeViewForEmptyCollection` could just take the string to show.